### PR TITLE
ENH: Add the ids='all'  as an option for mne.event.shift_time_events

### PR DESCRIPTION
**Pull request by Nichalas**
I added this option so that I can call `shift_time_events` in this manner:

```diff
- events = mne.event.shift_time_events( events, ids=np.unique(events[:,2]),  tshift=trigger_offset,  tshift, sfreq=raw.info['sfreq'])
+ events = mne.event.shift_time_events( events, ids='all',  tshift=trigger_offset,  tshift, sfreq=raw.info['sfreq'])
```

### CODE REVIEW

File: mne/event.py [hunk 1]
```diff
@@ -800,7 +800,7 @@ def shift_time_events(events, ids, tshift, sfreq):
     ----------
     events : array, shape=(n_events, 3)
         The events
-    ids : array int
+    ids : array int | 'all'
```
Comment (larsoner) [inline mne/event.py]: In other event things we use `None`, might as well do it here, too

File: mne/event.py [hunk 2]
```diff
@@ -814,8 +814,12 @@ def shift_time_events(events, ids, tshift, sfreq):
         The new events.
     """
     events = events.copy()
-    for ii in ids:
-        events[events[:, 2] == ii, 0] += int(tshift * sfreq)
+    if ids is None:
+        events[:, 0] += int(tshift * sfreq)
+    elif isinstance(ids, list):
+        for ii in ids:
+            events[events[:, 2] == ii, 0] += int(tshift * sfreq)
```
Comment (larsoner) [inline mne/event.py]: A nice DRY change would be to change the above branch mask=slice(None) and this one mask=np.in1d(events[:,2],IDs), then do the += below.

Comment (larsoner) [reply mne/event.py]: (on mobile so please forgive the errors, hopefully you get the idea)

**Comment by larsoner** [comment]
@Nichalas I pushed a few commits to clean things a bit. Can you add an entry to the `Changelog` in `whats_new.rst` (including a link to your URL at the bottom)?

**Comment by agramfort** [comment]
@Nichalas tell us if you approved these changes so we can merge. Thx

**Comment by Nichalas** [comment]
@agramfort yes, I approve!

**Comment by larsoner** [comment]
@Nichalas do you have a name/URL we should point to? We still need to update `whats_new.rst`

File: mne/tests/test_event.py [hunk 3]
```diff
@@ -542,4 +543,16 @@ def test_acqparser_averaging():
                         rtol=0, atol=1e-13)  # tol = 1 fT/cm
 
 
+def test_shift_time_events():
```
Comment (massich) [inline mne/tests/test_event.py]: it misses the docstring. 

```py
def test_shift_time_events():
    """Test shifting the events latency by a given amount."""
```

**Comment by massich** [comment]
thx @Nichalas !!

**Pull request merged**